src/qml/bitcoin.cpp

```diff
@@ -158,6 +160,188 @@ void setupChainQSettings(QGuiApplication* app, QString chain)
+bool m_isOnboarded;
+
+bool createNode(QGuiApplication& app, QQmlApplicationEngine& engine, int& argc, char* argv[], ArgsManager& gArgs)
+{
+    m_engine = &engine;
+
+    InitLogging(gArgs);
+    InitParameterInteraction(gArgs);
+
+    m_init = interfaces::MakeGuiInit(argc, argv);
+
+    m_node = m_init->makeNode();
+    m_chain = m_init->makeChain();
+
+    if (!m_node->baseInitialize()) {
+        // A dialog with detailed error will have been shown by InitError().
+        return EXIT_FAILURE;
+    }
+
+    m_handler_message_box.disconnect();
+
+    m_node_model = new NodeModel{*m_node};
+    m_executor = new InitExecutor{*m_node};
+    QObject::connect(m_node_model, &NodeModel::requestedInitialize, m_executor, &InitExecutor::initialize);
+    QObject::connect(m_node_model, &NodeModel::requestedShutdown, m_executor, &InitExecutor::shutdown);
+    QObject::connect(m_executor, &InitExecutor::initializeResult, m_node_model, &NodeModel::initializeResult);
+    QObject::connect(m_executor, &InitExecutor::shutdownResult, qGuiApp, &QGuiApplication::quit, Qt::QueuedConnection);
+
+    m_network_traffic_tower = new NetworkTrafficTower{*m_node_model};
+#ifdef __ANDROID__
+    AndroidNotifier android_notifier{*m_node_model};
+#endif
+
+    m_chain_model = new ChainModel{*m_chain};
+    m_chain_model->setCurrentNetworkName(QString::fromStdString(ChainTypeToString(gArgs.GetChainType())));
+    setupChainQSettings(m_app, m_chain_model->currentNetworkName());
+
+    QObject::connect(m_node_model, &NodeModel::setTimeRatioList, m_chain_model, &ChainModel::setTimeRatioList);
+    QObject::connect(m_node_model, &NodeModel::setTimeRatioListInitial, m_chain_model, &ChainModel::setTimeRatioListInitial);
+
+    qGuiApp->setQuitOnLastWindowClosed(false);
+    QObject::connect(qGuiApp, &QGuiApplication::lastWindowClosed, [&] {
+        m_node->startShutdown();
+    });
+
+    m_peer_model = new PeerTableModel{*m_node, nullptr};
+    m_peer_model_sort_proxy = new PeerListSortProxy{nullptr};
+    m_peer_model_sort_proxy->setSourceModel(m_peer_model);
+
+    m_engine->rootContext()->setContextProperty("networkTrafficTower", m_network_traffic_tower);
+    m_engine->rootContext()->setContextProperty("nodeModel", m_node_model);
+    m_engine->rootContext()->setContextProperty("chainModel", m_chain_model);
+    m_engine->rootContext()->setContextProperty("peerTableModel", m_peer_model);
+    m_engine->rootContext()->setContextProperty("peerListModelProxy", m_peer_model_sort_proxy);
+
+    m_options_model->setNode(&(*m_node), m_isOnboarded);
+
+    QObject::connect(m_options_model, &OptionsQmlModel::requestedShutdown, m_executor, &InitExecutor::shutdown);
+
+    m_engine->rootContext()->setContextProperty("optionsModel", m_options_model);
+
+    m_node_model->startShutdownPolling();
+
+    return true;
+}
+
+void startNodeAndTransitionSlot() { createNode(*m_app, *m_engine, m_argc, m_argv, gArgs); }
+
+// moved this function to a separate function for code reusability
```

Comment:
Comments like that seem add no value. However, this sentence is more suitable for a commit message.

Reply:
Removed. Thanks for pointing that out.